Fix Reports Theme

crossan007 opened the pull request:
Summary of changes in this PR:
- Reports Listing Page is themed:
![image](https://cloud.githubusercontent.com/assets/11679900/13801672/96fcb2a0-eb0a-11e5-8ae2-4b17324d3eb6.png)

- New Export CSV Theme:
![image](https://cloud.githubusercontent.com/assets/11679900/13656565/ee0c610a-e635-11e5-8f8e-8da6be28b6a3.png)

crossan007 commented:
New Export CSV Theme:
![image](https://cloud.githubusercontent.com/assets/11679900/13656565/ee0c610a-e635-11e5-8f8e-8da6be28b6a3.png)

crossan007 commented:
Query listings are now Select2:
![image](https://cloud.githubusercontent.com/assets/11679900/13801492/a9af7c54-eb08-11e5-993b-bde143abf75b.png)


And the parameters are dynamically rendered on the page:
![image](https://cloud.githubusercontent.com/assets/11679900/13801496/b552465e-eb08-11e5-859d-e9bf25703b00.png)


And so is the result data:
![image](https://cloud.githubusercontent.com/assets/11679900/13801499/c4688f9a-eb08-11e5-9f00-33513009728d.png)


Thoughts?

DawoudIO commented on churchinfo/js/RPCDummyAjax.js:
```diff
@@ -0,0 +1,19 @@
+        var group_ID = $('#GroupID option:selected').val();  // get the selected group ID
+       $.ajax({
+            method: "GET",
+            url:   window.CRM.root+"/api/groups/"+group_ID,
```
where does group_ID come from?

DawoudIO commented:
spacing still looks to be off in some files... 

the only Q i have is for sql injection testing and escaping chars in ReportService... I'll do some testing and see if i can break it.

DawoudIO commented:
So i noticed that all the box spacing is a bit off and that is due to the merge of 2 divs... please see the following change for fixing that

https://github.com/ChurchCRM/CRM/pull/402/commits/2ab9a9335ad1490823856482a7e240852db3933e

DawoudIO commented:
The filters require you to click the - to get a + then click the + to get the filter to show.

DawoudIO commented:
![image](https://cloud.githubusercontent.com/assets/554959/13908794/a6c2a9fe-eec7-11e5-803d-896549140b22.png)

Confirm Data Email gives a page error

crossan007 commented:
I think I've worked out all of the spacing bugs in this PR.

I addressed your concerns about SQL injection - although there are probably a bunch of other places in other files where this is an issue.

I also fixed the issue with sending all families emails from the LettersAndLabels report.

DawoudIO commented:
update and merge as you see fit.
